File: app/src/main/java/com/princeparadoxes/watertracker/data/sp/ProjectPreferenceModule.java
package com.princeparadoxes.watertracker.data.sp;

import android.app.Application;
import android.content.SharedPreferences;

import com.princeparadoxes.watertracker.ApplicationScope;

import dagger.Module;
import dagger.Provides;

import static android.content.Context.MODE_PRIVATE;

@Module
public class ProjectPreferenceModule {

    private static final String DEFAULT_SHARED_PREFERENCE_FILE_NAME = "SharedPreferencesDefault";

    @Provides
    @ApplicationScope
    SharedPreferences provideSharedPreferences(Application app) {
        return app.getSharedPreferences(DEFAULT_SHARED_PREFERENCE_FILE_NAME, MODE_PRIVATE);
    }

}

File: app/src/main/java/com/princeparadoxes/watertracker/data/source/sp/ProjectPreferenceModule.kt
package com.princeparadoxes.watertracker.data.source.sp

import android.app.Application
import android.content.SharedPreferences

import com.princeparadoxes.watertracker.ApplicationScope

import dagger.Module
import dagger.Provides

import android.content.Context.MODE_PRIVATE

@Module
class ProjectPreferenceModule {

    @Provides
    @ApplicationScope
    internal fun provideSharedPreferences(app: Application): SharedPreferences {
        return app.getSharedPreferences(DEFAULT_SHARED_PREFERENCE_FILE_NAME, MODE_PRIVATE)
    }

    companion object {

        private const val DEFAULT_SHARED_PREFERENCE_FILE_NAME = "SharedPreferencesDefault"
    }

}
